src/MemObject.h

```diff
@@ -167,6 +167,16 @@ class MemObject
 
     SwapOut swapout;
 
+    // IO for Rock::HeaderUpdater
+    class Update
+    {
+    public:
+        StoreIOState::Pointer reader; ///< reads old headers and old data
+        StoreIOState::Pointer writer; ///< writes new headers and old data
+    };
```

Comment:
I have several concerns about this particular solution:

* It is difficult for me to justify adding two low-level smart pointers (and associated dangerous destructor assertions) to _every_ MemObject for occasional and usually short-lived updates. Long-term, we should probably be moving SwapOut from MemObject rather than adding more SwapOut-like parts to MemObject.
* Moving private reader/writer pointers from the HeaderUpdater job (that tightly controls them) into public MemObject fields feels like a step backwards from design/encapsulation point of view.
* We want to mark StoreEntry for mgr:openfd_objects reports. Long-term, other code will have the same needs (e.g., if we start moving swapout state away from MemObject and/or if we add alternative ways to swap out entries "in the background"). It would be good to add a reusable mechanism that could address anticipated future needs instead of encouraging further MemObject pollution.

I suggest undoing this change and adding a MemObject::swapoutInterest data member. Ipc::StoreMapUpdate constructor/destructor will increment/decrement that member when locking/unlocking the entry. Any future swapout-related code will do the same. Cache manager will check for positive level when deciding whether to report the entry. Entry dumps will report swapout interest (when there is one).

If we can prove that two StoreMapUpdate objects cannot point to the same StoreEntry today, then the SwapoutInterest class can use a boolean field instead of a (larger) integer counter. If we have to use a counter, we may need to refactor Lock to reuse the corresponding level incrementing/decrementing code without Lock's virtual table pointer overheads.